lib/services/privileges_service.ts

```diff
@@ -1,192 +1,173 @@
 import { PureService } from '@Lib/services/pure_service';
-import { SNPredicate } from '@Models/core/predicate';
+import { SNLog } from '@Lib/log';
+import { SNNote } from '@Lib/models';
+import { SNProtocolService } from './protocol_service';
+import { SNStorageService } from '@Services/storage_service';
 import { StorageKey } from '@Lib/storage_keys';
-import { ContentType } from '@Lib/models';
-import { PrivilegeCredential, ProtectedAction, SNPrivileges } from '@Models/app/privileges';
+import { isNullOrUndefined } from '@Lib/utils';
 
-export enum PrivilegeSessionLength {
+enum PrivilegeSessionLengthSeconds {
   None = 0,
   FiveMinutes = 300,
   OneHour = 3600,
   OneWeek = 604800,
 }
 
-type CredentialAuthMapping = Partial<Record<PrivilegeCredential, string>>
-
-const CredentialsMetadata = {
-  [PrivilegeCredential.AccountPassword] : {
-    label: 'Account Password',
-    prompt: 'Please enter your account password.'
-  },
-  [PrivilegeCredential.LocalPasscode] : {
-    label: 'Application Passcode',
-    prompt: 'Please enter your application passcode.'
-  }
-};
+export function isValidPrivilegesSessionLength(number: unknown): boolean {
+  return (
+    typeof number === 'number' &&
+    Object.values(PrivilegeSessionLengthSeconds).includes(number)
+  );
+}
 
-const ActionsMetadata = {
-  [ProtectedAction.ManageExtensions]: {
-    label: 'Manage Extensions'
+export const PrivilegesSessionDurations = [
+  {
+    valueInSeconds: PrivilegeSessionLengthSeconds.None,
+    label: "Don't Remember",
   },
-  [ProtectedAction.ManageBackups]: {
-    label: 'Download/Import Backups'
+  {
+    valueInSeconds: PrivilegeSessionLengthSeconds.FiveMinutes,
+    label: '5 Minutes',
   },
-  [ProtectedAction.ViewProtectedNotes]: {
-    label: 'View Protected Notes'
+  {
+    valueInSeconds: PrivilegeSessionLengthSeconds.OneHour,
+    label: '1 Hour',
   },
-  [ProtectedAction.ManagePrivileges]: {
-    label: 'Manage Privileges'
+  {
+    valueInSeconds: PrivilegeSessionLengthSeconds.OneWeek,
+    label: '1 Week',
   },
-  [ProtectedAction.ManagePasscode]: {
-    label: 'Manage Passcode'
-  },
-  [ProtectedAction.DeleteNote]: {
-    label: 'Delete Notes'
-  },
-};
+];
 
 /**
- * Privileges allows certain actions within the application to require extra authentication.
- * For example, the privileges service exposes functions that allow the action of deleting
- * a note or viewing a note to require extra authentication.
- * Privileges are a superficial-level locking feature; they do not deal at all with underlying
- * data state. For example, viewing a protected note may require extra authentication,
- * but the underlying note data may already be decrypted in application memory, whether or not
- * the user has yet authenticated this action.
+ * Privileges enforce certain actions to require extra authentication,
+ * like viewing a protected note.
  */
 export class SNPrivilegesService extends PureService {
```

Comment:
I would be ok renaming this class to SNProtectionService, if you also dislike the word privileges as much as I do. But perhaps can be done in a followup PR so that the rename doesn't make git think it's a new file.

Reply:
I agree, the word privileges have a lot of unnecessary connotations here… I'll make this change in a separate PR. How about `SNAuthorizationService` instead, to align it with the methods in this file? (And also because it only protects something if the client is nice enough to call the methods 😄)

Reply:
That could work. `auth_service` is what used to be called the service to auth with the server. So it could go either way I guess. Could the remaining logic in the privileges_service just be grouped into the challenge_service? (Since ultimately a challenge is what's occurring or not occurring.)

Reply:
I'd like to keep the responsibility of handling authorization session length in a separate class, as that part is completely unrelated to the challenge_service